test/controllers/ImporterEORINumberControllerSpec.scala

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import base.ControllerSpecBase
+import controllers.actions.FakeDataRetrievalAction
+import forms.ImporterEORINumberFormProvider
+import mocks.repositories.MockSessionRepository
+import models.UserAnswers
+import pages.ImporterEORINumberPage
+import play.api.http.Status
+import play.api.mvc.{AnyContentAsFormUrlEncoded, Result}
+import play.api.test.FakeRequest
+import play.api.test.Helpers.{charset, contentType, defaultAwaitTimeout, redirectLocation, status}
+import views.html.ImporterEORINumberView
+
+import scala.concurrent.Future
+
+class ImporterEORINumberControllerSpec extends ControllerSpecBase {
+
+  val userAnswersWithImporterEORINumber: Option[UserAnswers] = Some(UserAnswers("some-cred-id")
+    .set(
+      ImporterEORINumberPage, "GB345834921000"
+    ).success.value
+  )
+
+  private def fakeRequestGenerator(importerEORI: String): FakeRequest[AnyContentAsFormUrlEncoded] =
+    fakeRequest.withFormUrlEncodedBody(
+      "importerEORI" -> importerEORI
+    )
+
+  trait Test extends MockSessionRepository {
+    lazy val controller = new ImporterEORINumberController(
+      authenticatedAction,
+      dataRetrievalAction,
+      dataRequiredAction,
+      mockSessionRepository,
+      messagesControllerComponents,
+      form,
+      ImporterEORINumberView
+    )
+    private lazy val ImporterEORINumberView = app.injector.instanceOf[ImporterEORINumberView]
+    private lazy val dataRetrievalAction = new FakeDataRetrievalAction(userAnswers)
+    val userAnswers: Option[UserAnswers] = Some(UserAnswers("some-cred-id"))
+    val formProvider: ImporterEORINumberFormProvider = injector.instanceOf[ImporterEORINumberFormProvider]
+    MockedSessionRepository.set(Future.successful(true))
+    val form: ImporterEORINumberFormProvider = formProvider
+  }
+
+  "GET onLoad" should {
+    "return OK" in new Test {
+      override val userAnswers: Option[UserAnswers] = Some(UserAnswers("credId"))
```

Comment:
this overrides     val userAnswers: Option[UserAnswers] = Some(UserAnswers("some-cred-id"))
which is the same empty user answers but with different id, what's the point here